hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -285,6 +285,16 @@ object LowerTableIR {
               )
             }
 
+        case TableDistinct(child) =>
+          val loweredChild = lower(child)
+
+          loweredChild.repartitionNoShuffle(loweredChild.partitioner.coarsen(child.typ.key.length).strictify)
+            .mapPartition { partition =>
+              mapIR(StreamGroupByKey(partition, child.typ.key)) { groupRef =>
+                ArrayRef(ToArray(StreamTake(groupRef, 1)), 0)
```

Comment:
Rather than going through `ToArray`, I would write this as a flatMap:
```
flatMapIR(StreamGroupByKey(partition, child.typ.key)) { groupRef =>
  StreamTake(groupRef, 1)
}
```
It's more direct and generates better code. (`ToArray` involves copying between regions, or will once stream memory management is in.)